repoman/flatpak.py

```diff
@@ -0,0 +1,386 @@
+gi.require_version('Gtk', '3.0')
+gi.require_version('Pango', '1.0')
+from gi.repository import Gtk, GObject, Pango
+
+import pyflatpak as flatpak
+
+import gettext
+gettext.bindtextdomain('repoman', '/usr/share/repoman/po')
+gettext.textdomain("repoman")
+_ = gettext.gettext
+
+class AddDialog(Gtk.Dialog):
+
+    remote_name = False
+
+    def __init__(self, parent):
+
+        settings = Gtk.Settings.get_default()
+        header = settings.props.gtk_dialogs_use_header
+
+        Gtk.Dialog.__init__(self, _("Add Source"), parent, 0,
+                            (Gtk.STOCK_CANCEL, Gtk.ResponseType.CANCEL,
+                             Gtk.STOCK_ADD, Gtk.ResponseType.OK),
+                             modal=1, use_header_bar=header)
+
+        self.log = logging.getLogger("repoman.FPAddDialog")
+
+
+        content_area = self.get_content_area()
+
+        content_grid = Gtk.Grid()
+        content_grid.set_margin_top(12)
+        content_grid.set_margin_left(12)
+        content_grid.set_margin_right(12)
+        content_grid.set_margin_bottom(12)
+        content_grid.set_row_spacing(6)
+        content_grid.set_column_spacing(6)
+        content_grid.set_halign(Gtk.Align.CENTER)
+        content_grid.set_hexpand(True)
+        content_area.add(content_grid)
+
+        add_title = Gtk.Label(_("Enter Source Details"))
+        Gtk.StyleContext.add_class(add_title.get_style_context(), "h2")
+        content_grid.attach(add_title, 0, 0, 2, 1)
+
+        self.name_entry = Gtk.Entry()
+        self.name_entry.set_placeholder_text(_("Source Name"))
+        self.name_entry.set_activates_default(True)
+        self.name_entry.connect(_("changed"), self.on_name_entry_changed)
+        self.name_entry.set_width_chars(20)
+        self.name_entry.set_margin_top(12)
+        content_grid.attach(self.name_entry, 0, 1, 1, 1)
+
+        self.url_entry = Gtk.Entry()
+        self.url_entry.set_placeholder_text(_("URL"))
+        self.url_entry.set_activates_default(True)
+        self.url_entry.connect(_("changed"), self.on_url_entry_changed)
+        self.url_entry.set_width_chars(50)
+        self.url_entry.set_margin_top(12)
+        content_grid.attach(self.url_entry, 1, 1, 1, 1)
+
+        self.add_button = self.get_widget_for_response(Gtk.ResponseType.OK)
+        self.add_button.set_sensitive(False)
+
+        self.cancel_button = self.get_widget_for_response(Gtk.ResponseType.CANCEL)
+        self.cancel_button.grab_focus()
+
+        Gtk.StyleContext.add_class(self.add_button.get_style_context(),
+                                   "suggested-action")
+        self.add_button.grab_default()
+
+        self.show_all()
+
+    def on_url_entry_changed(self, entry):
+        entry_text = entry.get_text()
+        entry_valid = flatpak.validate(entry_text)
+        try:
+            self.add_button.set_sensitive(entry_valid)
+        except TypeError:
+            pass
+    
+    def on_name_entry_changed(self, entry, data=None):
+        print(data)
```

Comment:
I think this line has to go as well. I'm still getting `None` in the console for every keystroke in the Name field.

Reply:
Line 106 that is. I'm not sure why it looks like I'm highlighting 3 previous lines as well.

Reply:
Good catch, thanks